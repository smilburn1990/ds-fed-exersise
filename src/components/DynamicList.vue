<template>
  <div>
    <lazy-component class="dynamic-list" v-for="(item, i) in data" :key="i">
      <div class="dynamic-list__left">
        <img class="dynamic-list__left--image" :src="item.image_src"/>
        <h3 class="dynamic-list__left--price">{{ item.price | currencyParse }}</h3>
        <h3 class="dynamic-list__left--price" :style="item.status === 'available' ? 'background-color:#04BAB3' : 'background-color:#ff4c4c'">{{ item.status }}</h3>
      </div>
      <div class="dynamic-list__right">
        <h2>{{ item.name }}</h2>
        <p>{{ item.desc }}</p>
      </div>
    </lazy-component>
  </div>
</template>

<script>
export default {
  name: 'DynamicList',
  props: {
    data: {
      type: Array,
      default: () => []
    }
  }
}
</script>

<style scoped lang="scss">
.dynamic-list {
  cursor: pointer;
  width: 100%;
  display: flex;
  padding: 2rem;
  background-color: #000;
  line-height: 1.7rem;
  color: #FFF;
  font-weight: 600;
  margin: 1rem 0;
  &:hover {
    transform: scale(1.01);
    opacity: 0.9;
  }
  &__left {
    padding: 1.5rem 0;
    width: 20%;
    text-align: center;
    &--image {
      width: 15rem;
    }
    &--price {
      text-align: center;
      margin: 0.3rem 0;
    }
  }
  &__right {
    padding: 0 2rem;
    width: 80%;
  }
}
</style>