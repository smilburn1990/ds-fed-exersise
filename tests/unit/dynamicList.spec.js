import { mount } from '@vue/test-utils'
import DynamicList from '@/components/DynamicList'

describe('DynamicList component', () => {
  const wrapper = mount(DynamicList)
  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })
  it('receives the correct props', () => {
    expect(typeof wrapper.vm.$options.props.data).toBe('object')
  })
  it("has the expected html structure", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
})
