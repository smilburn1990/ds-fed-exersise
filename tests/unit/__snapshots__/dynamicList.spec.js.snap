// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicList component has the expected html structure 1`] = `<div />`;
